scintillometry/io/psrfits/hdu.py

```diff
@@ -97,18 +123,43 @@ def frequency(self):
                          ((n_chan + 1) // 2)) * chan_bw
         return u.Quantity(freq, u.MHz, copy=False)
 
+    @frequency.setter
+    def frequency(self, freq):
+        """Frequency setter. The input should be the frequency array."""
+        freq = freq.to(u.MHz)
+        n_chan = len(freq)
+        # TODO how do we deal with the even number of nchan
+        c_chan = freq[int(n_chan / 2) + 1]
+        bw = freq.max() - freq.min()
+        self.hdu.header['OBSNCHAN'] = n_chan
+        self.hdu.header['OBSFREQ'] = c_chan
+        self.hdu.header['OBSBW'] = bw
+
     @property
     def ra(self):
         return Longitude(self.header['RA'], unit=u.hourangle)
 
+    @ra.setter
+    def ra(self, val):
+        self.hdu.header['RA'] = val.to_string(sep=':', pad=True)
+
     @property
     def dec(self):
         return Latitude(self.header['DEC'], unit=u.deg)
 
+    @dec.setter
+    def dec(self, val):
+        self.hdu.header['DEC'] = val.to_string(sep=':', alwayssign=True,
+                                               pad=True)
+
     @property
     def obs_mode(self):
         return self.header['OBS_MODE']
 
+    @obs_mode.setter
+    def obs_mode(self, val):
+        self.hdu.header['OBS_MODE'] = vale
```

Comment:
typo: `vale` - this means that you don't have a test case yet...  Also, here, I do think we should have a check that it is one of the allowed values...

Reply:
I am still trying to build the Writing API. So the test is not running yet.